projects/plugins/jetpack/extensions/shared/components/product-management-controls/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * WordPress dependencies
+ */
+import { BlockControls } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import ProductManagementInspectorControl from './inspector-control';
+import ProductManagementToolbarControl from './toolbar-control';
+import InvalidProductWarning from './invalid-product-warning';
+import StripeConnectToolbarButton from '../stripe-connect-toolbar-button';
+import { store as membershipProductsStore } from '../../../store/membership-products';
+
+import './style.scss';
+
+export default function ProductManagementControls( {
+	allowCreateOneTimeInterval = true,
+	selectedProductId = 0,
+	setSelectedProductId = () => {},
+} ) {
+	const products = useSelect(
+		select =>
+			select( membershipProductsStore ).getProducts( selectedProductId, setSelectedProductId ),
+		[]
+	);
+	const { connectUrl, isApiConnected, isSelectedProductInvalid, shouldUpgrade } = useSelect(
+		select => {
+			const { getConnectUrl, getShouldUpgrade, isApiStateConnected, isInvalidProduct } = select(
+				membershipProductsStore
+			);
+			return {
+				connectUrl: getConnectUrl(),
+				isApiConnected: isApiStateConnected(),
+				isSelectedProductInvalid: isInvalidProduct( selectedProductId ),
+				shouldUpgrade: getShouldUpgrade(),
+			};
+		}
+	);
+
+	if ( shouldUpgrade ) {
+		return null;
+	}
+
+	return (
+		<>
+			{ ! isApiConnected && !! connectUrl && (
+				<BlockControls group="block">
+					<StripeConnectToolbarButton blockName="premium-content" connectUrl={ connectUrl } />
```

Comment:
Was the blockName left hardcoded on purpose to "premium-content"?

Reply:
Nope, not on purpose. 😢